contracts/src/CrossChainApplications/NativeTokenBridge/INativeTokenDestination.sol

```diff
@@ -0,0 +1,45 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+/**
+ * @dev Interface that describes functionalities for a cross-chain native token bridge.
+ */
+interface INativeTokenDestination {
+    /**
+     * @dev Emitted when tokens are burned in this bridge contract to be bridged to the source chain.
+     */
+    event TransferToSource(
+        address indexed sender,
+        address indexed recipient,
+        uint256 amount,
+        address feeContractAddress,
+        uint256 feeAmount,
+        uint256 teleporterMessageID
+    );
+    /**
+     * @dev Emitted when tokens are not minted in order to collateralize the source contract.
+     */
+    event CollateralAdded(
+        uint256 amount,
+        uint256 remaining
+    );
+
+    /**
+     * @dev Emitted when minting bridge tokens.
+     */
+    event MintNativeTokens(address recipient, uint256 amount);
+
+    /**
+     * @dev This burns native tokens on this chain, and sends a message to the source
+     * chain to mint unlock tokens there.
```

Comment:
nit: extra mint, and not necessarily unlocking "tokens" right? Can be unlocking native assets on source chain?

Reply:
I think we use "native tokens" and "native assets" interchangeably.